### PR TITLE
Update some icons to address aliasing on high-res screens

### PR DESCRIPTION
**Pull request by richtabor**
Part of #https://github.com/WordPress/gutenberg/issues/49311, focusing on common blocks/UI icons first. Updates a few icons to remove the aliasing that occurs on high-resolution screens. 

## Why?
We want our icons to look crisp and present well.

## How?
Updating icons within the icons package. This does not change the spirit of the icons, just adjusts the icon paths a little to not use `0.25` X and Y values. 

## Testing Instructions
Check the icons in the following locations: 

1. Heading block. 
2. Gallery block.
3. Grid block.
4. Paragraph block.
5. Row block.
6. Stacked block.
7. List view icon.

## Screenshots or screencast 
CleanShot 2023-03-27 at 16 30 43
CleanShot 2023-03-27 at 16 31 11

### CODE REVIEW

**Comment by jasmussen** [comment]
Really stellar work. Are you also updating and publishing the Figma library as part of this? The gallery code looks slightly more expansive. Can you maybe run that through https://svgomg.net/?

**Comment by richtabor** [comment]
> The gallery code looks slightly more expansive. Can you maybe run that through https://svgomg.net/?

I did with one of the Figma plugins, ran it again. I can't seem to get it without `fill-rule` and `clip-rule` though. The icon comes out filled without those. And ideas how to reduce them?  

> Are you also updating and publishing the Figma library as part of this?

Yes! I have them ready, was waiting for approval first. I'll move these standouts into the actual icon components when we're merged here. 

CleanShot 2023-03-28 at 14 33 13

File: packages/icons/src/library/gallery.js
Comment (jasmussen) [inline packages/icons/src/library/gallery.js]: It looks like these rules are needed for the gallery icon, I just tested. But they should be `fillRule` and `clipRule` to be react native compatible.

**Review by jasmussen — APPROVED** [review]
This is an awesome attention to detail, thank you. Let's get this in.

Edit: just one small fix first.

File: packages/icons/src/library/gallery.js
Comment (richtabor) [reply packages/icons/src/library/gallery.js]: Good catch! Thanks!

**Pull request merged**
